catalog/video/catalog-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverAction.java

```diff
@@ -172,19 +149,30 @@ private String generateFilename() {
 
     private void updateParentWithChildMetadata(Metacard childMetacard)
             throws RolloverActionException {
-        parentMetacardUpdater.update(parentMetacard, childMetacard);
-        UpdateRequest updateRequest = createUpdateRequest(parentMetacard.getId(), parentMetacard);
-        submitParentUpdateRequest(updateRequest);
+        if (context.getParentMetacard()
+                .isPresent()) {
+            parentMetacardUpdater.update(context.getParentMetacard()
```

Comment:
Since you're using the parent metacard 3 times, it's probably best to create a local variable for it.